### PR TITLE
Vacations CSP sync

### PR DESCRIPTION
**Pull request by sponkrashin**
Issues #457, #459

### CODE REVIEW

File: server/Arcadia.Assistant.CSP/Model/Company.cs [hunk 1]
```diff
@@ -7,9 +7,9 @@ public partial class Company
     {
         public Company()
         {
-            this.CompanyHistory = new HashSet<CompanyHistory>();
```
Comment (santee) [inline server/Arcadia.Assistant.CSP/Model/Company.cs]: wtf

Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Model/Company.cs]: All models are auto-generated

File: server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                                    return new ApproveVacation.Success(result.CalendarEvent, result.Approvals.ToList());
+                                }
+
+                                return Calendar.Abstractions.EmployeeVacations.ApproveVacation.Success.Instance;
+                            },
+                            failure: err => new ApproveVacation.Error(err));
+                    break;
+
+                case CalendarEventRemovedFromApprovers msg:
+                    this.ApproveCalendarEvent(msg.Event.EventId)
```
Comment (santee) [inline server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: sure about that?

File: server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+            var timestamp = lastApproval?.Timestamp ?? DateTimeOffset.Now;
+
+            var newVacation = await this.vacationsSyncExecutor.UpdateVacation(newEvent, timestamp, updatedBy);
+
+            return new CalendarEventApprovedData(oldEvent, newVacation.CalendarEvent, updatedBy, timestamp);
+        }
+
+        private bool CheckDatesAvailability(CalendarEvent @event)
+        {
+            return true;
```
Comment (santee) [inline server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: supposed to?

Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: No, of course :)

File: server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs (hunk 2 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: No, this is the only part, that wasn't completely refactored yet.

Comment (santee) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: will you change it in this PR?

Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: Sure, I'm working on it right now.

Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: Done in much better way.

File: server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs (hunk 3 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: Done.

File: server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs [hunk 4]
```diff
@@ -6,28 +6,58 @@
...
+        private readonly ILoggingAdapter logger = Context.GetLogger();
+
         public CspEmployeeVacationsRegistry(
             VacationsSyncExecutor vacationsSyncExecutor,
             string employeeId)
         {
             this.vacationsSyncExecutor = vacationsSyncExecutor;
             this.employeeId = employeeId;
+
+            Context.System.Scheduler.ScheduleTellOnce(
```
Comment (santee) [inline server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: why not just Tell to self.

Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: I'm not sure I can just Tell in constructor.

Comment (santee) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: you can.

Comment (sponkrashin) [reply server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs]: Ok then, will fix it in next PR.

**Pull request merged**